src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -0,0 +1,637 @@
+import { FormEvent, useCallback, useMemo, useState } from 'react';
+
+import { NumberFormatValues } from 'react-number-format';
+import styled, { css } from 'styled-components';
+
+import { AlertType } from '@/constants/alerts';
+import { ButtonAction, ButtonShape, ButtonSize, ButtonType } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useURLConfigs } from '@/hooks/useURLConfigs';
+
+import { formMixins } from '@/styles/formMixins';
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { AlertMessage } from '@/components/AlertMessage';
+import { AssetIcon } from '@/components/AssetIcon';
+import { Button } from '@/components/Button';
+import { Checkbox } from '@/components/Checkbox';
+import { Details } from '@/components/Details';
+import { DiffOutput } from '@/components/DiffOutput';
+import { FormInput } from '@/components/FormInput';
+import { FormMaxInputToggleButton } from '@/components/FormMaxInputToggleButton';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Link } from '@/components/Link';
+import { Output, OutputType } from '@/components/Output';
+import { WithDetailsReceipt } from '@/components/WithDetailsReceipt';
+
+import { calculateCanViewAccount, calculateIsAccountViewOnly } from '@/state/accountCalculators';
+import { getSubaccount } from '@/state/accountSelectors';
+import { useAppSelector } from '@/state/appTypes';
+import { getUserVault } from '@/state/vaultSelectors';
+
+import { MustBigNumber } from '@/lib/numbers';
+
+const renderDiffOutput = ({
```

Comment:
[discussion] i feel like adding a layer of indirection isn't worth the extra typing we get here (if i'm reading this correctly). 

If we **really** want people to be intentional about how they use this component, i think we should add a docstring/comment to explain why we've restricting the props, cause it's not clear to me and if i were to add to this file i would likely just change this type to allow me to add whatever props i want

Reply:
Yeah I agree actually, I copied this pattern from a different file and I don't really understand why we do it